notesapp/api/serializers.py

```diff
@@ -0,0 +1,112 @@
+from rest_framework.serializers import ( 
+	ModelSerializer,
+	HyperlinkedIdentityField,
+	PrimaryKeyRelatedField,
+)
+from notesapp.model import KnowledgeBaseNotes, SoftSkillsDataNotes, RandomDataNotes
+						   
+
+
+knowledgebasenotes_detail_url = HyperlinkedIdentityField(
+    # URL to detail view
+    view_name='knowledgebasenotes-detail',
+    read_only=True
+)
+
+softskillsdatanotes_detail_url = HyperlinkedIdentityField(
+    # URL to detail view
+    view_name='softskillsdatanotes-detail',
+    read_only=True
+)
+
+randomdatanotes_detail_url = HyperlinkedIdentityField(
+    # URL to detail view
+    view_name='randomdatanotes-detail',
+    read_only=True
+)
+
+
+class KnowledgeBaseNotesSerializer(ModelSerializer):
+	"""
+	serializer class for KnowledgeBaseNotes model
+	"""
+	knowledge_base_notes = PrimaryKeyRelatedField(many=True, read_only=True)
+	url = knowledgebasenotes_detail_url
+
+	class Meta:
+		model = KnowledgeBaseNotes
+		fields = [
+			'url',
+			'id',
+			'user',
+			'resource',
+			'title',
+			'content',
+			'slug',
+			'created_at',
+			'modified_at',
+
+		]
+
+		extra_kwargs = {
+		'created_at' : {'read_only': True},
+		'modified_at' : {'read_only': True}
+
+		}
+
+
+class SoftSkillsDataNotesSerializer(ModelSerializer):
+	"""
+	serializer class for SoftSkillsDataNotes model
+	"""
+	soft_skills_data_notes = PrimaryKeyRelatedField(many=True, read_only=True)
+	url = softskillsdatanotes_detail_url
+
```

Comment:
there has to be a user PrimaryKeyRelatedField as well, as you need to take a user foreign key too
import the User model first